src/test/scala/org/clulab/habitus/variables/TestDefaultContextAttachment.scala

```diff
@@ -0,0 +1,104 @@
+package org.clulab.habitus.variables
+
+import org.clulab.habitus.utils.{DefaultContext, Test}
+import org.clulab.odin.Mention
+
+class TestDefaultContextAttachment extends Test {
+  // make sure a relation/event attachment is correct
+  // each test is supposed to have only one target mention extracted
+  val vp: VariableProcessor = VariableProcessor()
+
+  case class DefaultContextAttachmentTest(
+                           name: String,
+                           // Text for extraction
+                           text: String,
+                           label: String,
+                           desired: Map[String, String]
+                           // Expected attachment fields and values
+                         ) {
+
+    def getMentions(text: String): Seq[Mention] = {
+      val parsingResults = vp.parse(text)
+      // in this test, we only need the relation/event mentions, not text bound mentions
+      parsingResults.targetMentions
+    }
+
+
+    def test(index: Int): Unit = {
+      it should s"have a correct attachment in $name" in {
+        if (index == -1)
+          println("Put a breakpoint here to observe a particular test.")
+
+        // get event/relation mentions from text
+        val relevantMentions = getMentions(text).filter(_.label == label)
+        // there should be one target mention
+        // (relation or event) extracted
+        relevantMentions.length should be (1)
+        val attachments = relevantMentions.head.attachments
+        // there should be one attachment per target mention
+        attachments.toList.length should be (1)
+        val attachment = attachments.head.asInstanceOf[DefaultContext]
+        attachment.location should be (desired("location"))
+        attachment.date should equal (desired("date"))
+        attachment.process should equal (desired("process"))
+        attachment.crop should equal (desired("crop"))
+        attachment.fertilizer should equal (desired("fertilizer"))
+        attachment.comparative.toString should equal (desired("comparative"))
+      }
+    }
+  }
+
+  behavior of "VariableReader Entities"
+
+  val variableTests: Array[DefaultContextAttachmentTest] = Array(
+    DefaultContextAttachmentTest(
+      "sent-1",
+      "The area sown with rice and fertilized with urea for this 2021/2022 wintering campaign is 28,223 ha in wintering in Senegal.",
+      "PlantingArea",
+      Map(
```

Comment:
I like this use of a Map for this small number of tests.  If there were to be lots of them eventually, then the keys might be turned into constants/variables so that the compiler will check for typos or in case the name ever changes.  val DATE = "date".  If they all have the same keys, it could be a class instead, possibly with automatic defaults of missing values to "N/A".

Reply:
I did not completely understand about the class, but is what i did in this commit what you meant about variables? https://github.com/clulab/habitus/pull/137/commits/5277b4df279050e1dd7b3d119971183c8bcf44b5

Reply:
Yes.  If it was me, I would have made a typo in one of the keys already, but now the compiler will cover me.  The other way is something like

```scala
  case class Record(location: String = NA, date: String = NA, process: String = NA, crop: String = NA, fertilizer: String = NA, comparative: String = NA)

  Record(location = "Senegal", date = "2021/2022", process = "planting", comparative = "0")
```
which guarantees that the set of keys is fixed so that I can't add "oops" -> "a daisy" and the default value is automatic, .  However, it was mostly a thought experiment and I wouldn't change what's there without a better reason.

Reply:
I see! That does look nice...

Reply:
I know you said not to change it... But I changed it... It is just so much more beautiful this way!

Reply:
I have to learn to keep my mouth closed ;-)

Reply:
Nooo! I need to learn!! Thanks!!

Reply:
Does it look ok now? Should I merge?

Reply:
I was just waiting on Mihai since he was named as another reviewer and I wouldn't want to cross him.  Something in a different email hinted at a disagreement.  It's fine with me.